### PR TITLE
Adding seasonal kwargs to NaiveForecaster

### PR DESCRIPTION
**Pull request by Piyush1729**
Introduced seasonal  as boolean kwargs and refactored seasonal_last  #227

### CODE REVIEW

File: sktime/forecasting/naive.py [hunk 1]
```diff
@@ -20,26 +20,36 @@ class NaiveForecaster(OptionalForecastingHorizonMixin, BaseLastWindowForecaster)
 
     Parameters
     ----------
-    strategy : str{"last", "mean", "seasonal_last"}, optional (default="last")
+    seasonal : True or False or None, optional (default=None)
```
Comment (mloning) [inline sktime/forecasting/naive.py:23]: I think `seasonal` should be True or False (not None), default should be False. Or what did you have in mind with None?

File: sktime/forecasting/naive.py [hunk 2]
```diff
@@ -20,26 +20,36 @@ class NaiveForecaster(OptionalForecastingHorizonMixin, BaseLastWindowForecaster)
...
+
+    strategy : str{"last", "mean"}, optional (default="last")
         Strategy used to make forecasts:
 
         * "last": forecast the last value in the training series
         * "mean": forecast the mean of (a given window) of the training series
-        * "seasonal_last": forecasts the last value of the same season in the training series
-
+        *  seasonal is True and "last": forecasts the last value of the same season in the training series
+        * seasonal is True and "mean": forecast the mean (of a given window) of the same season in the training series
```
Comment (mloning) [inline sktime/forecasting/naive.py:34]: Yes, so the meaning of window changes slightly depending on whether seasonal is true or false:
* if seasonal=True, it refers to the number of previous seasons that we compute the mean over
* if seasonal=False, it refers to the number of previous time points we compute the mean over

File: sktime/forecasting/naive.py [hunk 3]
```diff
@@ -20,26 +20,36 @@ class NaiveForecaster(OptionalForecastingHorizonMixin, BaseLastWindowForecaster)
...
+    
     window_length : int or None, optional (default=None)
         Window length to use in the `mean` strategy. If None, entire training
             series will be used.
     """
 
-    def __init__(self, strategy="last", window_length=None, sp=None):
+    def __init__(self, strategy="last",seasonal=None,window_length=None, sp=None):
```
Comment (mloning) [inline sktime/forecasting/naive.py:41]: Please ensure that the code follows the PEP8 coding style

File: sktime/forecasting/naive.py (hunk 1 above)
Comment (Piyush1729) [reply sktime/forecasting/naive.py:23]: Will change it.

**Comment by mloning** [comment]
* Perhaps we don't even need the `seasonal` kwarg, could we not just use `sp` and compute the seasonal last/mean whenever `sp != 1`? 
* Some of the failing tests are unrelated to your changes, we're currently undergoing some refactoring, which should resolve these issues (see #246 for details)

**Comment by Piyush1729** [comment]
> * Perhaps we don't even need the `seasonal` kwarg, could we not just use `sp` and compute the seasonal last/mean whenever `sp != 1`?

okay, then we should remove boolean kwarg seasonal, right?

> * Some of the failing tests are unrelated to your changes, we're currently undergoing some refactoring, which should resolve these issues (see #246 for details)

I think unit test is to be added to test successfuly

**Comment by mloning** [comment]
Hi @Piyush1729 yes, let's remove the `seasonal` kwarg and just include `sp`. I'm happy if you add more unit tests!

**Comment by Piyush1729** [comment]
For seasonal mean, should the last window be whole training data? Else there might be a possibility where last window is less than sp which means that we cant forecast for all seasons.

**Comment by mloning** [comment]
@Piyush1729 yes good point! I think we should still allow window_length for the mean strategy. How about this: If for `strategy="mean"` both `window_length` and `sp` are given, we should raise an error if `window_length` is less than `sp`.

**Comment by Piyush1729** [comment]
@mloning I dont seem to understand why these tests are failing.

**Comment by mloning** [comment]
@Piyush1729, there are some merge conflicts that you need to resolve first. Try to 
* [sync your fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork) with the main repo, 
* pull the latest changes from dev,
* merge dev into your branch and resolve the conflicts. 

Hope this helps! :)

**Comment by Piyush1729** [comment]
@mloning I have synced my fork with repo. Now how do I rerun tests on the final commit?

**Comment by mloning** [comment]
Hi @Piyush1729, some merge conflicts haven't been resolved properly I think. We still need to clean up the PR a bit. I'm happy to help (I send you a pm)!

**Comment by mloning** [comment]
Closed in favour of #335

**Pull request closed without merging**